client/src/ui/molecules/collection/menu-v2.tsx

```diff
@@ -0,0 +1,244 @@
+  }, []);
+
+  useEffect(() => {
+    getItem(doc).then(setSavedItem);
+  }, [doc]);
+
+  useEffect(() => {
+    if (savedItem) setFormItem(savedItem);
+  }, [savedItem]);
+
+  useEffect(() => {
+    if (showNewCollection === false) {
+      setDisableAutoClose(false);
+    }
+  }, [showNewCollection]);
+
+  const collectionChangeHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = e.target;
+    if (value === addValue) {
+      setDisableAutoClose(true);
+      setShowNewCollection(true);
+    }
+    changeHandler(e);
+  };
+
+  const changeHandler = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    e.preventDefault();
+    const { name, value } = e.target;
+    setFormItem({ ...formItem, [name]: value });
+  };
+
+  const cancelHandler = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFormItem(savedItem || defaultItem);
+    setShow(false);
+  };
+
+  const saveHandler = async (
+    e: React.FormEvent<HTMLFormElement> | React.BaseSyntheticEvent
+  ) => {
+    e.preventDefault();
+    if (!collections) return;
+    await saveItem(formItem);
+    setSavedItem(formItem);
+    setShow(false);
+  };
+
+  const enterHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      saveHandler(e);
+    }
+  };
+
+  const deleteHandler = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (!collections) return;
+    await deleteItem(formItem);
+    setSavedItem(undefined);
+    setFormItem(defaultItem);
+    setShow(false);
+  };
+
+  return (
+    <DropdownMenuWrapper
+      className="watch-menu"
+      isOpen={show}
+      setIsOpen={setShow}
+      disableAutoClose={disableAutoClose}
+    >
+      {doc ? (
+        <Button
+          type="action"
+          isDisabled={isOffline}
+          icon={savedItem ? "bookmark-filled" : "bookmark"}
+          extraClasses={`bookmark-button small ${savedItem ? "highlight" : ""}`}
+          onClickHandler={() => {
+            setShow((v) => !v);
+          }}
+        >
+          <span className="bookmark-button-label">
+            {savedItem ? "Saved" : "Save"}
+          </span>
+        </Button>
+      ) : (
+        <Button
+          icon="edit"
+          type="action"
+          isDisabled={isOffline}
+          title="Edit"
+          onClickHandler={() => {
+            setShow((v) => !v);
+          }}
+        >
+          <span className="visually-hidden">Edit bookmark</span>
+        </Button>
+      )}
+      <form method="post" onSubmit={saveHandler}>
+        <DropdownMenu>
+          <div
+            className={`${menuId} show`}
+            role="menu"
+            aria-labelledby={`${menuId}-button`}
+          >
+            <button
+              onClick={cancelHandler}
+              className="watch-submenu-header mobile-only"
+            >
+              <span className="watch-submenu-header-wrap">
```

Comment:
Should we split from the watch-submenu styles?

Reply:
Yes, but this will require some changes in the main branch - I've added an issue to make those changes - merging this for now.